x-pack/platform/plugins/shared/stack_connectors/server/connector_types/openai/openai.ts

```diff
@@ -76,28 +125,137 @@ export class OpenAIConnector extends SubActionConnector<Config, Secrets> {
+    };
+    this.logger.debug(
+      `Provider: ${this.provider}, certificateFile: ${this.configAny.certificateFile}, certificateData: ${this.configAny.certificateData}, privateKeyFile: ${this.configAny.privateKeyFile}, privateKeyData: ${this.configAny.privateKeyData}`
+    );
+
+    try {
+      if (
+        this.provider === OpenAiProviderType.Other &&
+        (this.configAny.certificateFile ||
+          this.configAny.certificateData ||
+          this.configAny.privateKeyFile ||
+          this.configAny.privateKeyData)
+      ) {
+        // Validate PKI configuration
+        if (
+          !validatePKICertificates(
+            this.configAny.certificateFile,
+            this.configAny.certificateData,
+            this.configAny.privateKeyFile,
+            this.configAny.privateKeyData
+          )
+        ) {
+          throw new Error('Invalid or inaccessible PKI certificates');
+        }
+
+        let cert: string;
+        let key: string;
+
+        if (this.configAny.certificateFile) {
+          cert = fs.readFileSync(
+            Array.isArray(this.configAny.certificateFile)
+              ? this.configAny.certificateFile[0]
+              : this.configAny.certificateFile,
+            'utf8'
+          );
+        } else if (this.configAny.certificateData) {
+          cert = formatPEMContent(this.configAny.certificateData, 'CERTIFICATE', this.logger);
+        } else {
+          throw new Error('No certificate file or data provided');
+        }
+
+        if (this.configAny.privateKeyFile) {
+          key = fs.readFileSync(
+            Array.isArray(this.configAny.privateKeyFile)
+              ? this.configAny.privateKeyFile[0]
+              : this.configAny.privateKeyFile,
+            'utf8'
+          );
+        } else if (this.configAny.privateKeyData) {
+          key = formatPEMContent(this.configAny.privateKeyData, 'PRIVATE KEY', this.logger);
+        } else {
+          throw new Error('No private key file or data provided');
+        }
+
+        const httpsAgent = new https.Agent({
+          cert,
+          key,
+          rejectUnauthorized: this.configAny.verificationMode === 'none',
+          checkServerIdentity:
+            this.configAny.verificationMode === 'certificate' ||
+            this.configAny.verificationMode === 'none'
+              ? () => undefined
+              : undefined,
+        });
+
+        // Debug log the agent configuration
+        this.logger.debug('HTTPS Agent configuration:');
+        this.logger.debug(
+          JSON.stringify(
+            {
+              cert: cert.slice(0, 100) + '...',
+              key: key.slice(0, 100) + '...',
+              rejectUnauthorized: this.configAny.verificationMode === 'none',
+              checkServerIdentity:
+                this.configAny.verificationMode === 'certificate' ||
+                this.configAny.verificationMode === 'none',
             },
+            null,
+            2
+          )
+        );
```

Comment:
The debug logs for the HTTPS agent configuration include sensitive certificate and key data, even if truncated. This poses a security risk.

Please remove sensitive data from the debug logs.